src/backend/drm/compositor/mod.rs

```diff
@@ -1561,6 +1041,22 @@ where
     }
 }
 
+bitflags::bitflags! {
+    /// Possible flags for a DMA buffer
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    pub struct FrameMode: u32 {
+        /// Allow to realize the frame by compositing elements on the primary plane
+        const COMPOSITE = 1;
+        const PRIMARY_PLANE_SCANOUT = 2;
+        const OVERLAY_PLANE_SCANOUT = 4;
+        const CURSOR_PLANE_SCANOUT = 8;
+        /// Allow to realize the frame by assigning elements on planes
+        const SCANOUT = Self::PRIMARY_PLANE_SCANOUT.bits() | Self::OVERLAY_PLANE_SCANOUT.bits() | Self::CURSOR_PLANE_SCANOUT.bits();
+
+        const ALL = Self::COMPOSITE.bits() | Self::SCANOUT.bits();
```

Comment:
tbh I don't think this will fit in the current DrmCompositor API. This will require allowing custom assign algorithms. What I have in mind is that we make something similar to the FrameState public and actually externalize the whole assigning algorithm.

Reply:
Sounds good. (And is also not a very important problem atm.)